### DISCUSSION TITLE
Better MSBuild UX for users

### DISCUSSION THREAD

**Discussion by baronfel** [MoiraeSoftware/Myriad#167] [Ideas]
I was talking with the MSBuild team and I have an idea that might make Myriad much easier to use from an end-user perspective. Right now users have to generate files and add them to the `Compile` MSBuild item manually, ensuring correct ordering. However, if we know (or have a good idea) of what the final generated paths for Myriad-generated items will be, we could make a Task that basically recreates the Compile item array, but with the myriad-generated items in their correct positions, then replace the Compile item group with the outputs of this task wholesale. T he only major problem to solve would be finding the appropriate position to place this rewrite task - it would need to be after Myriad generation, but before the compilation, and it would also need to be before any tasks that add common/well-known MSBuild metadata to Compile items.

The implementation of the task would basically be `inputItems |> Seq.collect MapToMyriadOutputIfNecessary`, so the maintenance overhead would hopefully be minimal.

**Comment by 7sharp9** [comment]
Hi @baronfel I never got a notification for this so this is a long distance reply, sorry about that!

I did add some post/pre build events to Myriad so that it integrates more inot the project so you can discover whats in the compile list of items so what you describe somewhat fit in there.  I cant quite remember if thats in a branch or I merged it though...
